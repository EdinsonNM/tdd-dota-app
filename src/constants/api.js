export const OpenDotaApiUrl = {
    getPlayer: (id) => `players/${id}`,
};
