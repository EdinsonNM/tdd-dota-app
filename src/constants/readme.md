En esta secci√≥n se colocan constantes globales a todo el proyecto
