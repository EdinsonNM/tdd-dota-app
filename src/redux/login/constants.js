export const LOGIN_ACTION = "LOGIN_ACTION";
export const LOGIN_DONE_ACTION = "LOGIN_DONE_ACTION";
