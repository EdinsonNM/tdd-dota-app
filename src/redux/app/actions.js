import {createActions} from "redux-actions";
import {APP_ACTION} from "./constants";

const actionsCreator = createActions({}, APP_ACTION);
export const {appAction} = actionsCreator;
