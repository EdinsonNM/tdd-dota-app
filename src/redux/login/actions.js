import {createActions} from "redux-actions";
import {LOGIN_ACTION, LOGIN_DONE_ACTION} from "./constants";

const actionsCreator = createActions(
    {},

    LOGIN_ACTION,
    LOGIN_DONE_ACTION
);

export const {loginAction, loginDoneAction} = actionsCreator;
