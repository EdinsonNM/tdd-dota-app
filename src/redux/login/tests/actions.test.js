import {loginAction, loginDoneAction} from "../actions";
import {PlayerBuilder} from "../../../constants/mocks/player.mock";
import {LOGIN_DONE_ACTION} from "../constants";
describe("login/actions", () => {
    test("should create action LOGIN_ACTION", () => {
        expect(loginAction(1)).toMatchSnapshot();
    });
    test("should create action LOGIN_DONE_ACTION", () => {
        const player = PlayerBuilder();
        const expected = {
            type: LOGIN_DONE_ACTION,
            payload: player,
        };
        expect(loginDoneAction(player)).toEqual(expected);
    });
});
