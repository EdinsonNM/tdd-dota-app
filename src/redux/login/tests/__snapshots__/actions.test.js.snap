// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login/actions should create action LOGIN_ACTION 1`] = `
Object {
  "payload": 1,
  "type": "LOGIN_ACTION",
}
`;
